flow/src/main/java/flow/ActivityFlowSupport.java

```diff
@@ -0,0 +1,159 @@
+package flow;
+
+import android.os.Bundle;
+import java.util.Iterator;
+
+import static flow.Preconditions.checkArgument;
+import static flow.Preconditions.checkState;
+
+/**
+ * Manages a Flow within an Activity.  Make sure that each method is called from the corresponding
+ * method in the Activity.
+ * <p>
+ * Example:
+ *
+ * <pre>{@code
+ * public class MainActivity extends Activity {
+ *   private ActivityFlowSupport activityFlowSupport;
+ *   private final Flow.Dispatcher dispatcher = ...;
+ *
+ *   &#064;Override protected void onCreate(Bundle savedInstanceState) {
+ *     super.onCreate(savedInstanceState);
+ *     Parceler parceler = new GsonParceler();
+ *     Backstack defaultBackstack = Backstack.single(new MyAppIntroScreen());
+ *     ActivityFlowSupport.NonConfigurationInstance nonConfig =
+ *         (ActivityFlowSupport.NonConfigurationInstance) getLastNonConfigurationInstance();
+ *     flowSupport =
+ *         ActivityFlowSupport.onCreate(nonConfig, savedInstanceState, parceler, defaultBackstack);
+ *   }
+ *
+ *   &#064;Override public void onResume() {
+ *     super.onResume();
+ *     activityFlowSupport.onResume(flowDispatcher);
+ *   }
+ *
+ *   &#064;Override protected void onPause() {
+ *     activityFlowSupport.onPause();
+ *     super.onPause();
+ *   }
+ *
+ *   &#064;Override public Object onRetainNonConfigurationInstance() {
+ *     return activityFlowSupport.onRetainNonConfigurationInstance();
+ *   }
+ *
+ *   &#064;Override public void onSaveInstanceState(Bundle outState) {
+ *     super.onSaveInstanceState(outState);
+ *     activityFlowSupport.onSaveInstanceState(outState);
+ *   }
+ *
+ *   &#064;Override public void onBackPressed() {
+ *     if (!activityFlowSupport.onBackPressed()) {
+ *       super.onBackPressed();
+ *     }
+ *   }
+ *
+ *   &#064;Override public Object getSystemService(String name) {
+ *     Object service = activityFlowSupport.getSystemService(name);
+ *     return service != null ? service : super.getSystemService(name);
+ *   }
+ * }
+ * }</pre>
+ */
+public final class ActivityFlowSupport {
+  public static final class NonConfigurationInstance {
```

Comment:
Don't know what else we might want to shove in there.  Having a type instead of just Object makes the API a little clearer.